packages/orchestration/src/exos/stakingAccountKit.js

```diff
@@ -95,17 +108,49 @@ export const prepareStakingAccountKit = (baggage, makeRecorderKit, zcf) => {
         },
         // TODO move this beneath the Orchestration abstraction,
         // to the OrchestrationAccount provided by createAccount()
+        /**
+         * @param {string} [denom] - defaults to bondDenom
+         * @returns {Promise<{ amount: string; denom: string; }>}
+         */
+        async queryBalance(denom) {
+          const { chainAddress, queryConnection, bondDenom } = this.state;
+
+          denom ||= bondDenom;
+
+          const [result] = await E(queryConnection).query([
+            /** @type {RequestQueryJson} */ (
+              RequestQuery.toJSON(
+                RequestQuery.fromPartial({
+                  path: '/cosmos.bank.v1beta1.Query/Balance',
+                  data: QueryBalanceRequest.encode(
+                    QueryBalanceRequest.fromPartial({
+                      address: chainAddress,
+                      denom,
+                    }),
+                  ).finish(),
+                }),
+              )
+            ),
+          ]);
+          if (!result?.key) throw Fail`Error parsing result ${result}`;
+          const { balance } = QueryBalanceResponse.decode(
+            decodeBase64(result.key),
+          );
+          if (!balance) throw Fail`Error parsing result ${result}`;
```

Comment:
distinguish error message
```suggestion
          if (!balance) throw Fail`Result lacked balance: ${result}`;
```

Reply:
Even if the token doesn't exist, or the user has a 0 balance, we should still get a response:

https://github.com/Agoric/agoric-sdk/pull/9198/files#diff-591b38c7c82b1d2e45f9fd7ad844cd799e8242267e5290d8b7de8fe029934994R133-R139

If you are trying to say a `balance` key wasn't found, we could do that. But generally trying to indicate the decoding of the result failed.

Reply:
> If you are trying to say a balance key wasn't found, we could do that.

Yes

> But generally trying to indicate the decoding of the result failed.

Without `balance` it was a bad result, but the decoding didn't fail. It got through `decodeBase64` and `QueryBalanceResponse.decode` to return a value.